CMakeLists.txt

```diff
@@ -1398,15 +1405,12 @@ endif(ENABLE_BASH_SCRIPT_TESTING)
 # Create config.h file.
 configure_file("${netCDF_SOURCE_DIR}/config.h.cmake.in"
   "${netCDF_BINARY_DIR}/config.h")
-add_definitions(-DHAVE_CONFIG_H)
+
+add_compile_definitions(-DHAVE_CONFIG_H)
```

Comment:
Maybe for (yet) another PR, but is this actually needed at all? Are there circumstances when building netCDF that we *don't* have `config.h`? Is it even possible to build without it?

Reply:
There is no reason why we *shouldn't* have `config.h`, but it is good form to check for it regardless.  If it is absent for some reason (we get a lot of support email and unusual/'impossible' states aren't as rare as you'd hope, given the breadth of systems and environments in which netCDF is used), netCDF won't *currently* build, but that isn't to say it will always require `config.h` for every module.  If, in the future, there are circumstances for which `config.h` is not required (say, submodules or some other unforeseen situation), best to already have the "best practices" infrastructure in place.

Reply:
However, having it defined and required through a macro define is not a good design. You can use `__has_include` and `#error` to give the user a sensible error message if that is your intent with that.

And to clarify the issue regarding the snippet above is not about `config.h` it's about needing to define `-DHAVE_CONFIG_H` since we are always generating that file

Reply:
If `add_compile_definitions(-DHAVE_CONFIG_H)` is not defined, or defined only on the `netcdf` target, pretty much all of the plugins and other libraries netcdf builds will fail.

Reply:
Disregarding the design choice, it sounds like it should be `target_compile_definitions(PUBLIC)`. Not sure if it should be exported to the consumer as well or not, but you can scope it via generator expression ` $<BUILD_INTERFACE:...>` if needed.

Reply:
Slated for a future PR